util/file_reader_writer.cc

```diff
@@ -73,9 +73,14 @@ Status RandomAccessFileReader::Read(uint64_t offset, size_t n, Slice* result,
                                     char* scratch) const {
   Status s;
   uint64_t elapsed = 0;
+  // record the time spent waiting for ratelimiter, which should be excluded
+  // from elapsed when reporting
+  uint64_t total_wait_time = 0;
   {
-    StopWatch sw(env_, stats_, hist_type_,
-                 (stats_ != nullptr) ? &elapsed : nullptr);
+    StopWatchNano sw(env_);
+    if (env_) {
+      sw.Start();
```

Comment:
it's too confusing, I suspect now we're measuring time in cases we shouldn't like when stats are disabled. How about add a new function to `StopWatch` to track delay?